src/w11k-select.directive.ts

```diff
@@ -466,16 +466,26 @@ export function w11kSelect (w11kSelectConfig: Config,
         function setViewValue () {
           let selectedValues = internalOptions2externalModel(internalOptions, optionsExpParsed, w11kSelectConfig);
 
-          controller.$setViewValue(selectedValues);
+          let newViewValue: null|any[] = selectedValues;
+
+          if (scope.config.useNullableModel && selectedValues.length === 0) {
+            newViewValue = null;
+          }
+
+          controller.$setViewValue(newViewValue);
           updateHeader();
         }
 
         function updateNgModel () {
-          let value = internalOptions2externalModel(internalOptions, optionsExpParsed, w11kSelectConfig);
+          let value: null|any[] = internalOptions2externalModel(internalOptions, optionsExpParsed, w11kSelectConfig);
```

Comment:
@talikan since value is now nulled (before it was `any[]`) we need to add some nullchecks before accessing `value.length`. I'll add those and release a new version later today!

Reply:
Good catch, thank you!